src/webhook/mutation/pod_mutator/init_container.go

```diff
@@ -27,19 +28,37 @@ func createInstallInitContainerBase(webhookImage, clusterID string, pod *corev1.
 			{Name: config.K8sNamespaceEnv, ValueFrom: kubeobjects.NewEnvVarSourceForField("metadata.namespace")},
 			{Name: config.K8sNodeNameEnv, ValueFrom: kubeobjects.NewEnvVarSourceForField("spec.nodeName")},
 		},
-		SecurityContext: copyUserContainerSecurityContext(pod),
+		SecurityContext: securityContextForInitContainer(pod),
 		Resources:       *dynakube.InitResources(),
 	}
 }
 
-func copyUserContainerSecurityContext(pod *corev1.Pod) *corev1.SecurityContext {
-	var securityContext *corev1.SecurityContext
-	if len(pod.Spec.Containers) == 0 {
-		return securityContext
+func securityContextForInitContainer(pod *corev1.Pod) *corev1.SecurityContext {
+	var securityContext = &corev1.SecurityContext{
+		RunAsNonRoot:             address.Of(true),
+		ReadOnlyRootFilesystem:   address.Of(true),
+		AllowPrivilegeEscalation: address.Of(false),
+		Privileged:               address.Of(false),
+		Capabilities: &corev1.Capabilities{
+			Drop: []corev1.Capability{
+				"ALL",
+			},
+		},
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeRuntimeDefault,
+		},
 	}
-	if pod.Spec.Containers[0].SecurityContext != nil {
-		securityContext = pod.Spec.Containers[0].SecurityContext.DeepCopy()
+
+	var podSecurityContext = pod.Spec.Containers[0].SecurityContext
```

Comment:
when the container security context is empty, we have to also look at the pod security context and use the user/group from there if it is set.

Reply:
done.